compiler/rustc_hir_typeck/src/fn_ctxt/suggestions.rs

```diff
@@ -2869,6 +2861,19 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
                             return None;
                         }
 
+                        if checked_ty.is_box() && self.is_need_parenthesize(expr) {
```

Comment:
I presume due to the `is_box()`, the following code still leads to an invalid suggestion?

```rs
fn main() {
    let x = std::rc::Rc::new(Some(1)); 
    let test: Option<i32> = x as std::rc::Rc<Option<i32>>;
}
````

I guess you can remove that box check?

Reply:
> I guess you can remove that box check?

Thank you.  Updated.